syft/rekor/utils.go

```diff
@@ -0,0 +1,175 @@
+package rekor
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/anchore/syft/syft/file"
+	"github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/spdx/tools-golang/tvloader"
+)
+
+var (
+	//SbomPredicateType string = "google.com/sbom"
+	GoogleSbomPredicateType string = "http://lumjjb/sbom-documents"
+)
+
+type digest struct {
+	Sha256 string
+}
+
+type sbomEntry struct {
+	Format string
+	Digest digest
+	Uri    string
+}
+
+type buildMetadata struct {
+	ArtifactSourceRepo             string `json:"artifact-source-repo,omitempty"`
+	ArtifactSourceRepoCommit       string `json:"artifact-source-repo-commit omitempty"`
+	AttestationGeneratorRepo       string `json:"attestation-generator-repo,omitempty"`
+	AttestationGeneratorRepoCommit string `json:"attestation-generator-repo-commit,omitempty"`
+}
+
+// Predicate type = "google.com/sbom"
+type GoogleSbomPredicate struct {
+	Sboms         []sbomEntry
+	BuildMetadata buildMetadata `json:"build-metadata,omitempty"`
+}
+
+type InTotoAttestation struct {
+	in_toto.StatementHeader
+	Predicate GoogleSbomPredicate
+}
+
+type sbomWithDigest struct {
+	sha1 string
+	spdx *spdx.Document2_2
+}
+
+func pprintStruct(v any) string {
+	//json, err := json.MarshalIndent(v, "", "\t")
+	json, err := json.Marshal(v)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return string(json)
+}
+
+func parsePEMCert(decodedCert string) (*x509.Certificate, error) {
```

Comment:
You may want to use https://cs.github.com/sigstore/sigstore/blob/bdd7ba405f0967f97d560b81a157b1c48333f57e/pkg/cryptoutils/certificate.go#L82

` UnmarshalCertificatesFromPEMLimited(pemBytes, 1)` here instead of writing it yourself!